src/main/java/org/truffleruby/core/binding/BindingNodes.java

```diff
@@ -136,6 +151,12 @@ public static FrameSlotAndDepth findFrameSlotOrNull(String identifier, Materiali
         return null;
     }
 
+    public static FrameSlotAndDepth findFrameSlotOrNull(String identifier, FrameDescriptor frameDescriptor) {
```

Comment:
This method should have another name if it assumes the slot will always be found.